### PR TITLE
[13.0] auth_oauth_multi_token improvements

### PR DESCRIPTION
**Pull request by sbidoul**
Individual commits should make sense.

The main change is that old tokens are deleted instead of disabled.

@simahawk @fmdl @Yajo as you have contributed to this module, do you remember a reason why tokens are disabled and never deleted (they accumulate forever) ?

### CODE REVIEW

**Comment by sbidoul** [comment]
I am testing this with `auth_oidc` (#261). It works well, as far as the underlying Odoo oauth implementation is acceptable (Odoo never detects expired tokens for instance).

File: auth_oauth_multi_token/models/auth_oauth_multi_token.py [hunk 1]
```diff
@@ -30,24 +33,17 @@ def create(self, vals):
...
 
     def _oauth_validate_multi_token(self):
         """Check current user's token and clear them if max number reached."""
         user_tokens = self._oauth_user_tokens(self.user_id.id)
         max_token = self.user_id.oauth_access_max_token
         if user_tokens and len(user_tokens) > max_token:
             # clear last token
-            user_tokens[max_token - 1]._oauth_clear_token()
-
-    def _oauth_clear_token(self):
```
Comment (simahawk) [inline auth_oauth_multi_token/models/auth_oauth_multi_token.py:49]: I'd keep `_oauth_clear_token` and delete them there. Offers a hook and keeps backward compat.

**Comment by fmdl** [comment]
@sbidoul just for log

**Review by yajo — APPROVED** [review]
I don't remember why it was like it was, but this makes sense.

**Review by simahawk — APPROVED** [review]

File: auth_oauth_multi_token/models/auth_oauth_multi_token.py (hunk 1 above)
Comment (sbidoul) [reply auth_oauth_multi_token/models/auth_oauth_multi_token.py:49]: good point, done

**Pull request merged**